### PR TITLE
Use yewdux for store and yew-router for routes

### PR DESCRIPTION
**Pull request by ecashin**

### CODE REVIEW

File: nextspeaker-web/src/components.rs [hunk 1]
```diff
@@ -1,30 +1,92 @@
...
+            selected.value = new_selection;
+        }
+    });
+    html! {
+        <button {onclick}>{"choose"}</button>
+    }
+}
+
+#[derive(Properties, PartialEq)]
+
```
Comment (bbenne10) [inline nextspeaker-web/src/components.rs:81]: eh?

File: nextspeaker-web/src/components.rs [hunk 2]
```diff
@@ -1,30 +1,92 @@
...
+pub struct SimulationPanelProps {}
+
+#[styled_component]
+pub fn SimulationPanel(_props: &SimulationPanelProps) -> Html {
+    html! {
+        <div>
```
Comment (bbenne10) [inline nextspeaker-web/src/components.rs:87]: Unless you *need* this `<div>`, you can do `<>` to avoid adding an additional element to the dom.

**Comment by bbenne10** [comment]
I was on a call while I looked at this and I just realized this is already merged :P

**Comment by ecashin** [comment]
Thank you for the interesting comments, though.

**Comment by bbenne10** [comment]
I think only one of the comments is ACTUALLY approaching interesting, but I appreciate the kindness. 😁

**Pull request merged**